### PR TITLE
Updated IBM/Lenovo OS definitions for proper detection + LLDP Mappings

### PR DESCRIPTION
**Pull request by fz883**
Please give a short description what your pull request is for:

IBM sold their Networking business to Lenovo back in ~2015 or so. With that Lenovo inherited switches like the G8052 and others that, starting with FW version 8.x were rebranded to Lenovo. The mibs are the same, but the device are actually Lenovo branded. In order to rebrand those switches in LibreNMS to Lenovo to distinguish between them, I created a new OS called "enos" - [link](https://www.lenovo.com/medias/lenovo-enterprise-networking-os-ds.pdf?context=bWFzdGVyfC9lbWVhL3BkZi9zaXRlbWFwL3wxMzM5MzN8YXBwbGljYXRpb24vb2N0ZXQtc3RyZWFtfC9lbWVhL3BkZi9zaXRlbWFwL2hlZC9oNzYvOTM4OTAxNjI4NTIxNC5wZGZ8ZGNmYWYyZTc3MzhhYjFkN2QyZTgyMTYwZDRiMjQ0YWQzZThhOWM0Zjk0MDVlZTkxZDE2NWZlMGU0MWQyZGY1Nw). Before that, Lenovo rebranded IBM switches were shown as generic devices in the device overview of LibreNMS with no OS info, no FW version, no serial info.

The CNOS (pure Lenovo FW) definition was updated to gather FW version + serial. After that, the overview shows all device types correctly.

image

I also updated the discovery-protocols.php file to incorporate some changes needed in order to get LLDP mappings correct. Before that some links from CNOS to ENOS devices were not "linked" to the remote switch & port because the ports were not found in the DB. If the changes I made are in the wrong spot, please advise. After the change the LLDP and network mappings are fine, the generated map shows all devices correctly linked.

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by fz883** [comment]
ok, Travis fails because snmprec files are not available. please advise on those,

File: includes/polling/os/cnos.inc.php [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ *
+ * Copyright (c) 2020 Florian Zillner <zillo83@gmail.com>
+ * Parts Copyright (c) 2015 Søren Friis Rosiak <sorenrosiak@gmail.com>
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+ */
+$sysdescr_value = $device['sysDescr'];
```
Comment (Jellyfrog) [inline includes/polling/os/cnos.inc.php:13]: these should be converted to the new OS format, see for example https://github.com/librenms/librenms/blob/master/LibreNMS/OS/Endrun.php

File: includes/polling/os/enos.inc.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
+<?php
```
Comment (Jellyfrog) [inline includes/polling/os/enos.inc.php:1]: dito

**Review by Jellyfrog — CHANGES_REQUESTED** [review]

**Comment by Jellyfrog** [comment]
I don't think we need all those MIBs

**Comment by Jellyfrog** [comment]
it also fails on syntax, https://travis-ci.com/github/librenms/librenms/jobs/375363635

**Comment by murrant** [comment]
Yes, standard mibs should not be included in vendor directories.

**Comment by Jellyfrog** [comment]
Thank you for your contribution but we're closing this due to inactivity.  
Feel free to reopen if you plan on finishing it!

**Pull request closed without merging**